src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+      setPolarValues(_total_weights, a, p, weight);
+    }
+  }
+}
+
+
+/**
+ * @brief Converts this Quadrature to a character array of its attributes.
+ * @details The character array includes the number of polar angles, the
+ *          the values of the sine and weight of each polar angle, and the
+ *          product of the sine and weight of each polar angle.
+ * @return a character array of the Quadrature's attributes
+ */
+std::string Quadrature::toString() {
+
+  std::stringstream string;
+
+  string << "Quadrature";
+  string << "\n\t# azim angles  = " << _num_azim;
+  string << "\n\t# polar angles = " << _num_polar;
+
+  string << "\n\tphis = ";
+  if (_phis != NULL) {
+    for (int a = 0; a < _num_azim/4; a++)
+      string << _phis[a] << ", ";
+  }
+
+  string << "\n\tazim weights = ";
+  if (_azim_weights != NULL) {
+    for (int a = 0; a < _num_azim/4; a++)
+      string << _azim_weights[a] << ", ";
+  }
+
+  string << "\n\tthetas = ";
+  if (_thetas != NULL) {
+    for (int a = 0; a < _num_azim/4; a++) {
+      for (int p = 0; p < _num_polar/2; p)
+        string << " (" << a << "," << p << "): " << _thetas[a][p] << ", ";
+    }
+  }
+
+  string << "\n\tpolar weights = ";
+  if (_polar_weights != NULL) {
+    for (int a = 0; a < _num_azim/4; a++) {
+      for (int p = 0; p < _num_polar/2; p)
+        string << " (" << a << "," << p << "): " << _polar_weights[a][p] << ", ";
+    }
+  }
+
+  string << "\n\tsin thetas = ";
+  if (_sin_thetas != NULL) {
+    for (int a = 0; a < _num_azim/4; a++) {
+      for (int p = 0; p < _num_polar/2; p)
+        string << " (" << a << "," << p << "): " << _sin_thetas[a][p] << ", ";
+    }
+  }
+
+  string << "\n\ttotal weights = ";
+  if (_total_weights != NULL) {
+    for (int a = 0; a < _num_azim/4; a++) {
+      for (int p = 0; p < _num_polar/2; p)
+        string << " (" << a << "," << p << "): " << _total_weights[a][p] << ", ";
+    }
+  }
+
+  return string.str();
+}
+
+
+quadratureType Quadrature::getQuadratureType() {
+  return _quad_type;
+}
+
+
+/**
+ * @brief Dummy constructor calls the parent constructor.
+ */
+TYPolarQuad::TYPolarQuad(): Quadrature() {
+  _quad_type = TABUCHI_YAMAMOTO;
+}
+
+
+/**
+ * @brief Set the number of polar angles to initialize.
+ * @param num_polar the number of polar angles (maximum 3)
+ */
+void TYPolarQuad::setNumPolarAngles(const int num_polar) {
+
+  if (num_polar > 6)
+    log_printf(ERROR, "Unable to set the number of polar angles to %d "
+               "for TYPolarQuad (max 6 angles)", num_polar);
+
+  Quadrature::setNumPolarAngles(num_polar);
+}
+
+
+/**
+ * @brief Routine to initialize the polar quadrature.
+ * @details This routine uses the tabulated values for the Tabuchi-Yamamoto
+ *          polar angle quadrature, including the sine thetas and weights.
+ */
+void TYPolarQuad::initialize() {
+
+  /* Call parent class initialize routine */
+  Quadrature::initialize();
+
+  /* Allocate temporary arrays for tabulated quadrature values */
```

Comment:
This should have a conditional to check if `thetas` was already initialized or not. If it is not `NULL` then it should first be deallocated and then re-allocated. These sorts of deallocation issues creep up and bite for multi-simulation situations.